From 587430dc804bf991e1175cbff25288b18f64aa1a Mon Sep 17 00:00:00 2001
From: xeniape <xenia.fischer@stackable.tech>
Date: Tue, 23 Sep 2025 09:31:00 +0200
Subject: Stop building unused extensions.

By default Druid builds all community extensions and then discards them
while assembling the final distribution. This patch removes unused
extensions from the build.
---
 pom.xml | 30 ------------------------------
 1 file changed, 30 deletions(-)

diff --git a/pom.xml b/pom.xml
index 29f0c12301..b291470096 100644
--- a/pom.xml
+++ b/pom.xml
@@ -230,38 +230,8 @@
         <module>extensions-core/druid-testcontainers</module>
         <!-- Community extensions -->
         <module>extensions-contrib/compressed-bigdecimal</module>
-        <module>extensions-contrib/influx-extensions</module>
-        <module>extensions-contrib/cassandra-storage</module>
-        <module>extensions-contrib/dropwizard-emitter</module>
-        <module>extensions-contrib/cloudfiles-extensions</module>
-        <module>extensions-contrib/graphite-emitter</module>
-        <module>extensions-contrib/distinctcount</module>
-        <module>extensions-contrib/druid-exact-count-bitmap</module>
-        <module>extensions-contrib/statsd-emitter</module>
-        <module>extensions-contrib/time-min-max</module>
-        <module>extensions-contrib/virtual-columns</module>
-        <module>extensions-contrib/thrift-extensions</module>
-        <module>extensions-contrib/ambari-metrics-emitter</module>
-        <module>extensions-contrib/sqlserver-metadata-storage</module>
         <module>extensions-contrib/kafka-emitter</module>
-        <module>extensions-contrib/redis-cache</module>
-        <module>extensions-contrib/opentsdb-emitter</module>
-        <module>extensions-contrib/materialized-view-maintenance</module>
-        <module>extensions-contrib/materialized-view-selection</module>
-        <module>extensions-contrib/momentsketch</module>
-        <module>extensions-contrib/moving-average-query</module>
-        <module>extensions-contrib/tdigestsketch</module>
-        <module>extensions-contrib/ddsketch</module>
-        <module>extensions-contrib/influxdb-emitter</module>
-        <module>extensions-contrib/gce-extensions</module>
-        <module>extensions-contrib/aliyun-oss-extensions</module>
         <module>extensions-contrib/prometheus-emitter</module>
-        <module>extensions-contrib/opentelemetry-emitter</module>
-        <module>extensions-contrib/grpc-query</module>
-        <module>extensions-contrib/druid-iceberg-extensions</module>
-        <module>extensions-contrib/druid-deltalake-extensions</module>
-        <module>extensions-contrib/spectator-histogram</module>
-        <module>extensions-contrib/rabbit-stream-indexing-service</module>
         <!-- distribution packaging -->
         <module>distribution</module>
         <!-- Revised integration tests -->
